src/Avalonia.Controls/ItemsControl.cs

```diff
@@ -280,6 +286,16 @@ protected virtual void OnContainersRecycled(ItemContainerEventArgs e)
         {
         }
 
+        /// <summary>
+        /// Gets or sets a value which indicates whether to wrap around when the first
+        /// or last item is reached.
+        /// </summary>
+        public bool WrapSelection
```

Comment:
I did start implementing this myself a while ago but aborted it because i came across the problem that I'm seeing here: `WrapSelection` is being defined on `ItemsControl` which doesn't actually support selection as such.

The problem is that selection is added by the derived `SelectingItemsControl`, but kinda piggy-backs of focus as things currently stand. This is actually incorrect and may need changing at some point in future (on WPF you can generally move focus without changing selection by using Ctrl+Arrow) but means that "selection" behavior probably shouldn't be defined on `ItemsControl`.

Maybe exposing the property on `SelectingItemsControl` and adding a `private protected WrapFocus { get; set; }` property to `ItemsControl` might be a good solution for now?